skietbaan_fe/src/components/LeaderboardPage.js

```diff
@@ -1,143 +1,325 @@
 import React, { Component } from 'react';
-import Select from 'react-select';
+import { connect } from 'react-redux';
+import { fetchleaderboadfilterdata } from '../actions/postActions';
+import { fetchleaderboadtabledata } from '../actions/postActions';
+import { Collapse } from 'react-collapse';
+import { Table } from "react-bootstrap";
+import { getCookie } from './cookie.js'
+import { MDBBtn, MDBIcon } from "mdbreact";
+import {
+    BrowserView,
+    MobileView,
+    isBrowser,
+    isMobile
+} from "react-device-detect";
+
 import '../bootstrap/LeaderboardStyle.css';
+/* import '../bootstrap/NavbarMenuStyle.css'; */
+import { runInThisContext } from 'vm';
 
 
 
 class LeaderboardPage extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            user: {},
-            groups: [],
-            competitions: [],
+            names: ["name 1",
+                "name 2",
+                "name 3",
+                "name 4",
+                "name 5",
+                "name 6",
+                "name 7",],
             selectedGroup: "1",
             selectedCompetition: "1",
-            selectedST: "1",
-            rankingTable: []
+            selectedScoreType: "1",
+            selectedRank: null,
+            collapseFilter: false
         }
         this.setCompetitionValue = this.setCompetitionValue.bind(this);
         this.setGroupValue = this.setGroupValue.bind(this);
         this.setScoreTypeValue = this.setScoreTypeValue.bind(this);
-        this.updateLeaderboard = this.updateLeaderboard.bind(this);
+        this.getLeaderboardData = this.getLeaderboardData.bind(this);
+        this.onMouseClickFilter = this.onMouseClickFilter.bind(this);
+        this.print = this.print.bind(this);
+        this.displayScoreByType = this.displayScoreByType.bind(this);
+        this.getCurentUserRankNumber = this.getCurentUserRankNumber.bind(this);
+        this.showMoreScores = this.showMoreScores.bind(this);
     }
     //executed when leaderboard in mounted on main app
     componentWillMount() {
-        fetch('http://localhost:63474/api/Leaderboard/GetLeaderboardFilterData?UserID=' + 1)
-            .then(res => res.json())
-            .then(data => this.setState({ groups: data.groups1, competitions: data.competitions1, id: data.id }, console.log(data.competitions), console.log(data.groups), console.log(this.state.groups)));
-            this.updateLeaderboard();
+        //get function to get filter data from api
+        this.props.fetchleaderboadfilterdata();
+        this.getLeaderboardData();
+    }
+    getLeaderboardData() {
+        let token = getCookie("token"); 
+        console.log("displaying tokken")
+        console.log(token);
+        const filterSelection = {
+            selectedCompetition: this.state.selectedGroup,
+            selectedGroups: this.state.selectedCompetition,
```

Comment:
why selectedCompetitions: this.state.selectedgroup, should it not be this.state.selectedCompetition?